src/sprite_weapon.cpp

```diff
@@ -0,0 +1,169 @@
+		if (!anim) {
+			Output::Warning("Invalid battler animation ID {}", battler->GetBattleAnimationId());
+			return;
+		}
+
+		const auto* battler_animation_weapon = lcf::ReaderUtil::GetElement(anim->weapons, weapon_animation_id);
+		if (!battler_animation_weapon) {
+			return;
+		}
+
+		SetSrcRect(Rect(frame * 64, battler_animation_weapon->weapon_index * 64, 64, 64));
+	}
+
+	const bool flip = battler->IsDirectionFlipped();
+	SetFlipX(flip);
+}
+
+void Sprite_Weapon::SetWeaponAnimation(int nweapon_animation_id) {
+	weapon_animation_id = nweapon_animation_id;
+}
+
+void Sprite_Weapon::StartAttack(bool secondary_weapon) {
+	cycle = 0;
+
+	attacking = true;
+
+	SetZ(default_z + (secondary_weapon ? 1 : -1));
+
+	auto* battler = GetBattler();
+
+	if (battler->GetBattleAnimationId() > 0) {
+		const lcf::rpg::BattlerAnimation* anim = lcf::ReaderUtil::GetElement(lcf::Data::battleranimations, battler->GetBattleAnimationId());
+		if (!anim) {
+			Output::Warning("Invalid battler animation ID {}", battler->GetBattleAnimationId());
+			return;
+		}
+
+		const auto* battler_animation_weapon = lcf::ReaderUtil::GetElement(anim->weapons, weapon_animation_id);
+		if (!battler_animation_weapon) {
+			return;
+		}
+
+		StringView sprite_file = battler_animation_weapon->weapon_name;
+
+		if (!sprite_file.empty()) {
+			FileRequestAsync* request = AsyncHandler::RequestFile("BattleWeapon", sprite_file);
+			request->SetGraphicFile(true);
+			request_id = request->Bind(&Sprite_Weapon::OnBattleWeaponReady, this, battler_animation_weapon->weapon_index);
+			request->Start();
+		}
+	}
+}
+
+void Sprite_Weapon::StopAttack() {
+	attacking = false;
+}
+
+void Sprite_Weapon::CreateSprite() {
+	auto* battler = GetBattler();
+
+	SetX(battler->GetDisplayX());
+	SetY(battler->GetDisplayY());
+
+	// Not animated -> Monster
+	SetOx(32);
+	SetOy(32);
+	ResetZ();
+}
+
+void Sprite_Weapon::OnBattleWeaponReady(FileRequestResult* result, int32_t weapon_index) {
+	SetBitmap(Cache::Battleweapon(result->file));
+	SetSrcRect(Rect(0, weapon_index * 64, 64, 64));
+}
+
+void Sprite_Weapon::Draw(Bitmap& dst) {
+	if (!attacking) {
+		return;
+	}
+
+	auto* battler = GetBattler();
+
+	SetTone(Main_Data::game_screen->GetTone());
+	SetX(battler->GetDisplayX());
+	SetY(battler->GetDisplayY());
+	SetFlashEffect(battler->GetFlashColor());
+
+	Sprite_Battler::Draw(dst);
+}
+
+void Sprite_Weapon::ResetZ() {
+	static_assert(Game_Battler::Type_Ally < Game_Battler::Type_Enemy, "Game_Battler enums re-ordered! Fix Z order logic here!");
+
+	constexpr int id_limit = 128;
+
+	const auto& graphic = GetBitmap();
+	int y = battler->GetBattlePosition().y;
+	if (battler->GetType() == Game_Battler::Type_Enemy && graphic) {
+		y += graphic->GetHeight() / 2;
+	}
+
+	int z = battler->GetType();
+	z *= SCREEN_TARGET_HEIGHT * 2;
+	z += y;
+	z *= id_limit;
+	z += id_limit - battle_index;
+	z += Priority_Battler;
+
+	default_z = z;
+
+	SetZ(z);
```

Comment:
Is there dead code in the Sprite_Weapon class? This looks like a copy-paste of the old Sprite_Battler class.
There are comments and checks concerning enemies in here. Enemies cannot use CBA, so I'm confused.

Reply:
Fixed. The enemy code has been removed.